src/app/engine/audio-backend.service.ts

```diff
@@ -110,11 +110,16 @@ export class AudioBackendService {
   }
 
   private loadBank(url: string) {
+    // on Safari we need to use callbacks using decodeAudioData method.
     this.http
       .get(url, { responseType: 'arraybuffer' })
-      .pipe(mergeMap((result) => this.context.decodeAudioData(result)))
+      .pipe(mergeMap((result) => new Promise((resolve) => {
+        this.context.decodeAudioData(result, (buffer) => {
```

Comment:
👍

This can be further simplified (and also handle errors properly) by changing to 
 `new Promise((resolve, reject) => this.context.decodeAudioData(result, resolve, reject))`

Reply:
@urish yes, thanks.
Code simplified, and handle error added